src/plugins/data/public/ui/saved_query_flyouts/open_saved_query_flyout.tsx

```diff
@@ -302,7 +302,9 @@ export function OpenSavedQueryFlyout({
                 if (selectedQuery) {
                   if (
                     // Template queries are not associated with data sources. Apply data source from current query
-                    selectedQuery.attributes.isTemplate
+                    selectedQuery.attributes.isTemplate ||
+                    // Associating a saved query with a data source is optional. If no data source is present, keep the current source.
+                    !selectedQuery.attributes.query.dataset?.dataSource
```

Comment:
Will depend on other folks to validate correctness, but is there a reason we're not covering this case with tests given its an edge case we wouldn't want to regress?

Reply:
+1 on the test. 

Also If we are making this change, Id like to first understand why `queryStringManager.setQuery(updatedQuery);` needs to be called here. Previously, this codepath was only invoked for saved query templates, so it was something we could look into later given the impact radius. Now with this change, even regular saved queries will use this code path. If thats the case, I want to understand why we needed to set the `queryStringManager.setQuery(updatedQuery)` in the first place when we didnt need to do this before

Reply:
Refactored code to not store dataset and also ignore it when loading saved queries